### ISSUE TITLE
[Feature] Implement async API calls to Cloudflare

### ISSUE THREAD

**Issue by Stewart86** [cloudflare/python-cloudflare#121]
With increase popularity of async programming with Python, we are trying to update our code base to be as much as possible be asynchronous in our IO bound processes. As we use Cloudflare API intensively, it will be super helpful if Cloudflare official python package can be async ready so that we do not have to result to building our own client for Cloudflare APIs or some hackish wrapper function for `Cloudflare` calls. 

Having browsed through the code in this repository, I think I can take up the challenge to extend this package to support async API call to Cloudflare using `aiohttp` or `httpx`. with a call like:

```python
async def print_zones():
    cf = CloudFlare.AsyncCloudFlare()
    # or 
    cf = CloudFlare.CloudFlare(async=True)
    zones = await cf.zones.get()
        for zone in zones:
            zone_id = zone['id']
            zone_name = zone['name']
            print zone_id, zone_name
```

However, one down side is, Python version would have to be greater than 3.7 to reduce any compatibility issues with third party async http packages.

**Comment by mahtin** [comment]
Yes, quite doable. A good idea. 

For reference: https://docs.python-requests.org/en/v0.8.3/user/advanced/

**Comment by Stewart86** [comment]
The reference you mentioned does not use Python native `asyncio` implementation. It uses [gevent](http://www.gevent.org/) and it only allow batch requests but not providing non-blocking code. 

From `requests` [documentation](https://docs.python-requests.org/en/stable/user/advanced/#blocking-or-non-blocking), it also mentioned that it does not provide non-blocking IO calls.

> Requests does not provide any kind of non-blocking IO. 

Also, the at the same topic, the document recommends that:

>  there are lots of projects out there that combine Requests with one of Python’s asynchronicity frameworks. Some excellent examples are requests-threads, grequests, requests-futures, and httpx.

**Comment by Stewart86** [comment]
I have took a deep look at the current code. Unfortunately to implement async support, it seems to me that many, if not most, put of the entire code base would require some modification.

Seeing that no tests were done, such refactoring would be very risky. As such, I believe that re-writing would make more sense. 

So I took a weekend to rebuild this API with reference to many of the logic you have used in this code base, refactored, and modernised and to Python best practices. 

Feel free to check it out and your comment would be very much appreciated. https://github.com/Stewart86/aioCloudflare

At this point, I would also be closing this issue.
